torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -24,6 +25,38 @@ static constexpr char* kEngineCPUQueueSize =
     "local_autograd_engine_cpu_queue_size";
 static constexpr char* kNumAutogradContexts = "num_autograd_contexts";
 
+struct AccumulateGradPreHook : GraphTask::GraphTaskFunctionPreHook {
+  AccumulateGradPreHook(
+      std::shared_ptr<AccumulateGrad> accumulateGrad,
+      ContextPtr autogradContext)
+      : accumulateGrad_(std::move(accumulateGrad)),
+        autogradContext_(std::move(autogradContext)) {}
+
+  void operator()(const variable_list& grads) override {
+    torch::autograd::check_input_variables("AccumulateGrad", grads, 1, 0);
+    const auto& grad = grads[0];
+    // It is possible that the grad is not defined since a separate
+    // invocation of the autograd engine on the same node might actually
+    // compute this gradient. Also accumulate grads only for
+    // AccumulateGrad function.
```

Comment:
"Also accumulate grads only for AccumulateGrad function". This part of the comment is not valid here?

Reply:
deleted.